tests/system-tests/internal/supporttools/traceroute.go

```diff
@@ -0,0 +1,441 @@
+package supporttools
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/namespace"
+
+	"github.com/openshift-kni/eco-gotests/tests/system-tests/internal/apiobjectshelper"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/deployment"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+	scc "github.com/openshift-kni/eco-gotests/tests/system-tests/internal/scc"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	supportToolsDeployRBACName = "privileged-rdscore-supporttools"
+	supportToolsDeploySAName   = "default"
+	supportToolsRBACRole       = "system:openshift:scc:privileged"
+)
+
+// CreateTraceRouteDeployment creates support-tools deployment on the nodes specified in scheduleOnNodes.
+func CreateTraceRouteDeployment(
+	apiClient *clients.Settings,
+	stNamespace,
+	stDeploymentLabel,
+	stImage string,
+	scheduleOnNodes []string) (*deployment.Builder, error) {
+	glog.V(100).Infof("Getting the image and tag for tools from the release image")
+
+	stDeploymentName := fmt.Sprintf("%s-support-tools", stNamespace)
+
+	glog.V(100).Infof("Create support-tools namespace %s", stNamespace)
+
+	err := ensureNamespaceExists(apiClient, stNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create support-tools namespace %s: %w", stNamespace, err)
+	}
+
+	glog.V(100).Infof("Adding SCC privileged to the support-tools namespace")
+
+	err = scc.AddPrivilegedSCCtoDefaultSA(stNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add SCC privileged to the supporttools namespace %s: %w",
+			stNamespace, err)
+	}
+
+	stDeployment, err := createTraceRouteDeployment(
+		apiClient,
+		stImage,
+		stDeploymentName,
+		stNamespace,
+		stDeploymentLabel,
+		scheduleOnNodes,
+	)
+
+	if err != nil {
+		return stDeployment, err
+	}
+
+	retries := 3
+	pollInterval := 5
+
+	for i := 0; i < retries; i++ {
+		stDeployment, err = deployment.Pull(apiClient, stDeploymentName, stNamespace)
+		if err != nil {
+			return nil, err
```

Comment:
this _return_ doesn't give a chance for another retry(if available) so maybe worth using _continue_ here?